packages/xdl/src/credentials/AndroidCredentials.js

```diff
@@ -0,0 +1,255 @@
+/* @flow */
+
+import path from 'path';
+
+import fs from 'fs-extra';
+import chalk from 'chalk';
+import crypto from 'crypto';
+import uuidv4 from 'uuid/v4';
+import spawnAsync from '@expo/spawn-async';
+import axios from 'axios';
+import ProgressBar from 'progress';
+import { spawn as ptySpawn } from 'node-pty-prebuilt';
+
+import { getCredentialsForPlatform } from './Credentials';
+import logger from '../Logger';
+import UserSettings from '../UserSettings';
+
+const NEWLINE = process.platform === 'win32' ? '\r\n' : '\n';
+const javaExecutable = process.platform === 'win32' ? 'java.exe' : 'java';
+
+export type Credentials = {
+  keystore: string,
+  keystorePassword: string,
+  keyPassword: string,
+  keystoreAlias: string,
+};
+
+export async function backupExistingCredentials(
+  { outputPath, username, experienceName }: Object,
+  log: any = logger.info.bind(logger),
+  logSecrets: boolean = true
+) {
+  const credentialMetadata = { username, experienceName, platform: 'android' };
+
+  log(`Retreiving Android keystore for ${experienceName}`);
+
+  const credentials = (await getCredentialsForPlatform(credentialMetadata): any);
+  if (!credentials) {
+    throw new Error('Unable to fetch credentials for this project. Are you sure they exist?');
+  }
+  const { keystore, keystorePassword, keystoreAlias: keyAlias, keyPassword } = credentials;
+
+  const storeBuf = Buffer.from(keystore, 'base64');
+  log(`Writing keystore to ${outputPath}...`);
+  fs.writeFileSync(outputPath, storeBuf);
+  if (logSecrets) {
+    log('Done writing keystore to disk.');
+    log(`Save these important values as well:
```

Comment:
Highlight this section with chalk so the developer knows they need to pay attention to this message.